File: Task_1/task_1.swift
import Foundation

func getRiverCrossingResult() -> Int {
    print("Введите общее количество валунов (stones): ", terminator: "")
    let stones = Int(readLine()!)! // Читаем количество валунов

    print("Введите количество птиц (N): ", terminator: "")
    let n = Int(readLine()!)!     // Читаем количество птиц

    var birds = [Int]()
    print("Введите дальность полета для каждой из \(n) птиц через пробел: ", terminator: "")
    let birdDistances = readLine()!.split(separator: " ").map { Int($0)! } // Читаем дальности полета
    for distance in birdDistances {
        birds.append(distance) // Заполняем массив
    }

    var visitedStones = [Bool](repeating: false, count: stones) // Создаем массив для отслеживания посещенных валунов.
                                                             // Изначально все значения false

    // Моделируем перелеты птиц
    for range in birds {
        for i in (range - 1)..<stones where i < visitedStones.count {
            visitedStones[i] = true // Помечаем валун как посещенный
        }
    }

    var unvisitedCount = 1
    for visited in visitedStones {
        if !visited {         // Если валун не посещен
            unvisitedCount += 1 // Увеличиваем счетчик
        }
    }
    if unvisitedCount == 1 {
        unvisitedCount = 0
    }

    return unvisitedCount // Возвращаем количество непосещенных валунов
}

let result = getRiverCrossingResult() // Вызываем функцию
print("Количество валунов, на которых не появится ни одной птицы: \(result)") // Выводим результат

File: Task_1/task_1.cs
using System;
using System.Linq;
public class RiverCrossing
{
    // Функция для ввода данных и вычислений
    static int GetRiverCrossingResult()
    {
        Console.WriteLine("Введите общее количество валунов (stones):");
        int stones = int.Parse(Console.ReadLine());
        Console.WriteLine("Введите количество птиц (N):");
        int n = int.Parse(Console.ReadLine());
        int[] birds = new int[n];
        Console.WriteLine($"Введите дальность полета для каждой из {n} птиц через пробел:");
        string[] birdDistances = Console.ReadLine().Split(' ');
        for (int i = 0; i < n; i++)
        {
            birds[i] = int.Parse(birdDistances[i]);
        }
        // Создаем массив, чтобы отслеживать посещенные валуны
        bool[] visitedStones = new bool[stones];
        // Моделируем перелет каждой птицы
        foreach (int range in birds)
        {
            for (int i = range - 1; i < stones; i += range)
            {
                visitedStones[i] = true;
            }
        }
        // Подсчитываем количество непосещенных валунов
        int unvisitedCount = 0;
        for (int i = 0; i < stones; i++)
        {
            if (!visitedStones[i])
            {
                unvisitedCount++;
            }
        }
        return unvisitedCount;
    }
    public static void Main(string[] args)
    {
        // Получаем результат вычислений
        int result = GetRiverCrossingResult();
        // Выводим результат
        Console.WriteLine($"Количество валунов, на которых не появится ни одной птицы: {result}");
    }
}
